src/quote/getQuoteAndBuyCoverInputs.ts

```diff
@@ -0,0 +1,161 @@
+  DEFAULT_COMMISSION_RATIO,
+  DEFAULT_SLIPPAGE,
+  MINIMUM_COVER_PERIOD,
+  NEXUS_MUTUAL_DAO_TREASURY_ADDRESS,
+  TARGET_PRICE_DENOMINATOR,
+} from '../constants/buyCover';
+import {
+  Address,
+  CoverRouterQuoteResponse,
+  ErrorApiResponse,
+  GetQuoteApiResponse,
+  GetQuoteResponse,
+  IntString,
+  Integer,
+  PoolCapacity,
+} from '../types';
+
+dotenv.config();
+
+type CoverRouterQuoteParams = {
+  productId: Integer;
+  amount: IntString;
+  period: Integer;
+  coverAsset: CoverAsset;
+};
+
+async function getQuoteAndBuyCoverInputs(
+  productId: Integer,
+  coverAmount: IntString,
+  coverPeriod: Integer, // days
+  coverAsset: CoverAsset,
+  coverBuyerAddress: Address,
+  paymentAsset: CoverAsset = coverAsset,
+  slippage: number = DEFAULT_SLIPPAGE, // 0.1%
+  ipfsData: string = '', // IPFS base32 hash value
+): Promise<GetQuoteApiResponse | ErrorApiResponse> {
+  let result: GetQuoteResponse | undefined = undefined;
+
+  if (!process.env.COVER_ROUTER_URL) {
+    return { result, error: { message: 'Missing COVER_ROUTER_URL env var', data: {} } };
+  }
+
+  if (!Number.isInteger(productId) || productId <= 0) {
+    return { result, error: { message: 'Invalid productId: must be a positive integer', data: {} } };
+  }
+
+  if (!/^\d+$/.test(coverAmount) || parseInt(coverAmount, 10) <= 0) {
+    return { result, error: { message: 'Invalid coverAmount: must be a positive integer string', data: {} } };
+  }
+
+  if (!Number.isInteger(coverPeriod) || coverPeriod < MINIMUM_COVER_PERIOD) {
+    return {
+      result,
+      error: { message: `Invalid coverPeriod: must be at least ${MINIMUM_COVER_PERIOD} days`, data: {} },
+    };
+  }
+
+  if (!Object.values(CoverAsset).includes(coverAsset)) {
+    return { result, error: { message: 'Invalid coverAsset', data: {} } };
+  }
+
+  if (!Object.values(CoverAsset).includes(paymentAsset)) {
+    return { result, error: { message: 'Invalid paymentAsset', data: {} } };
+  }
+
+  if (!/^0x[a-fA-F0-9]{40}$/.test(coverBuyerAddress)) {
+    return { result, error: { message: 'Invalid coverBuyerAddress: must be a valid Ethereum address', data: {} } };
+  }
+
+  if (typeof slippage !== 'number' || slippage < 0 || slippage > 1) {
+    return { result, error: { message: 'Invalid slippage: must be a number between 0 and 1', data: {} } };
+  }
+
+  if (ipfsData !== '' && !/^[A-Za-z2-7]{46}$/.test(ipfsData)) {
+    return { result, error: { message: 'Invalid ipfsData: must be a valid IPFS base32 hash value', data: {} } };
+  }
+
+  try {
+    const { quote, capacities } = await getQuote(productId, coverAmount, coverPeriod, coverAsset);
+
+    const maxPremiumInAsset = calculatePremiumWithCommissionAndSlippage(
+      BigInt(quote.premiumInAsset),
+      DEFAULT_COMMISSION_RATIO,
+      slippage,
+    );
+
+    result = {
+      displayInfo: {
+        premiumInAsset: quote.premiumInAsset,
+        coverAmount,
+        yearlyCostPerc: (BigInt(quote.annualPrice) / BigInt(TARGET_PRICE_DENOMINATOR)).toString(),
```

Comment:
🤔  Are these display info values meant to be used as is? I'm asking because in our UI we display the cover fee (premiumInAsset) and yearly cost including commission + slippage - so in case these are the final values for display, both the `premiumInAsset` and `yearlyCostPerc` should be wrapped in `calculatePremiumWithCommissionAndSlippage` to get the final display value.

Reply:
yes, we should make sure they include commission + slippage

Reply:
fixed